libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx

```diff
@@ -160,52 +181,41 @@ function GrantAccess() {
                 id: 'global:nationalId',
                 defaultMessage: 'Kennitala',
               })}
+              error={errors.toNationalId?.message}
               onChange={(value) => {
                 requestDelegation(value)
               }}
             />
           </GridColumn>
-          <GridColumn paddingBottom={2} span={['12/12', '12/12', '6/12']}>
-            <Controller
-              control={control}
-              name="name"
-              defaultValue=""
-              render={({ onChange, value, name }) => (
-                <Input
-                  name={name}
-                  readOnly={true}
-                  label={formatMessage({
-                    id:
-                      'service.portal.settings.accessControl:grant-label-user',
-                    defaultMessage: 'Aðfangshafi',
-                  })}
-                  placeholder={formatMessage({
-                    id:
-                      'service.portal.settings.accessControl:grant-placeholder-user',
-                    defaultMessage: 'Nafn',
-                  })}
-                  value={value}
-                  hasError={errors.name}
-                  errorMessage={errors.name?.message}
-                  onChange={onChange}
-                />
-              )}
-            />
+        </GridRow>
+        <GridRow>
+          <GridColumn span={['12/12', '12/12', '8/12', '12/12', '8/12']}>
+            <Box display="flex" justifyContent="flexEnd">
+              <Button
+                type="submit"
+                icon="arrowForward"
+                disabled={!name || loading}
+                loading={loading}
+              >
+                {name
+                  ? `${formatMessage({
+                      id:
+                        'service.portal.settings.accessControl:grant-form-pre-submit-active',
+                      defaultMessage: 'Veita',
+                    })} ${name} ${formatMessage({
+                      id:
+                        'service.portal.settings.accessControl:grant-form-post-submit-active',
+                      defaultMessage: 'aðgang',
+                    })}`
+                  : formatMessage({
+                      id:
+                        'service.portal.settings.accessControl:grant-form-submit-disabled',
+                      defaultMessage: 'Sláðu inn kennitölu aðgangshafa',
+                    })}
```

Comment:
Should use interpolated translations to help with i18n, but these labels aren't great (long, name not correct "fall") and think we should keep "Áfram".

We should either keep the disabled input (and wait for disabled design to be improved), or just render the name standalone below or to right of the ssn input.

Reply:
Ahh nice, didn't know interpolated translations were supported.

Yeah, rendering something based on user interaction is always difficult in regards to UX.
At first I just rendered the name as a standalone, both below and to the right of the input; and IMO it was a bad UX and just didn't look good. The focus of the user would either be on the button or the input, so I went with the button. I'm open for other ideas, but as I said, the plain text (or key/value text) was not a good option.
Agree that "fall" breaks that, not sure if rephrasing the text makes it better:

Reply:
Yeah, sorry, I just don't think this change is much better than how it is now. Having duplicate instructions in the inactive button, and an Icelandic issue in the active button.

I think we should split these changes up and get more eyes on this or a new design proposal.

Reply:
Yeah I agree